dmutils/content_loader.py

```diff
@@ -344,32 +346,72 @@ def _has_assurance(self, key):
         return bool(question) and question.get('assuranceApproach', False)
 
 
-class ContentLoader(object):
-    def __init__(self, manifest, content_directory):
-        manifest_sections = read_yaml(manifest)
-
-        self._questions = {
-            q: _load_question(q, content_directory)
-            for section in manifest_sections
-            for q in section["questions"]
-        }
-
-        self._sections = [
-            self._populate_section(s) for s in manifest_sections
-        ]
-
-    def get_question(self, question):
-        return self._questions.get(question, {}).copy()
+class ContentNotFoundError(Exception):
+    pass
 
-    def get_builder(self):
-        return ContentBuilder(self._sections)
 
-    def _populate_section(self, section):
-        section["id"] = _make_section_id(section["name"])
-        section["questions"] = [
-            self.get_question(q) for q in section["questions"]
+class ContentLoader(object):
+    """
+    Usage:
+    >>> loader = ContentLoader('path/to/digitalmarketplace-frameworks')
+    >>> # pre-load manifests
+    >>> loader.load_manifest('framework-1', 'manifest-1', 'question-set-1')
+    >>> loader.load_manifest('framework-1', 'manifest-2', 'question-set-1')
+    >>> loader.load_manifest('framework-1', 'manifest-3', 'question-set-2')
+    >>> loader.load_manifest('framework-2', 'manifest-1', 'question-set-1')
+    >>> # get a builder
+    >>> loader.get_builder(framework_slug, 'manifest-1')
+    """
+    def __init__(self, content_path):
+        self.content_path = content_path
+        self._content = defaultdict(dict)
+        # A defaultdict that defaults to a defaultdict of dicts
+        self._questions = defaultdict(partial(defaultdict, dict))
+
+    def get_builder(self, framework_slug, manifest):
+        try:
+            return ContentBuilder(self._content[framework_slug][manifest])
+        except KeyError:
+            raise ContentNotFoundError("Content not found for {} and {}".format(framework_slug, manifest))
+
+    def load_manifest(self, framework_slug, manifest, question_set):
```

Comment:
I'd go with `framework_slug, question_set, manifest` argument order, since manifests can only cover questions from one question set, and this would highlight the hierarchical order